File: client_electron/renderer/js/models/battleship/ship.js
export class Ship {
    constructor(id, type, size){
        this.id = id;
        this.type = type;
        this.size = size;
        // 0-> noActivo, 1->Activo, 2->Eliminado
        this.status = 0; 
        this.coords = [];
        this.attacked_cells = 0; 
    }

    set_coords(coords){
        this.coords = coords;
        this.status = 1; 
    }

    still_active(){
        this.attacked_cells ++;
        if(this.attacked_cells >= this.size){
            this.status = 2; 
            return true;
        }
    }
}

File: server_room/models/game/battleship/ship.py
class Ship:
    def __init__(self, id, type, coords, size):
        self.id = id
        self.type = type
        self.coords = coords  
        self.size = size
        self.attacked = 0
        self.status = 0  # 0 = activo, 1 = destruido

    def attack(self, x, y):
        for coord in self.coords:
            if coord.x == x and coord.y == y:
                self.attacked += 1
                if self.attacked == self.size:
                    self.status = 1
                return True
        return False
